lib/rubygems/remote_fetcher.rb

```diff
@@ -109,9 +109,12 @@ def api_endpoint(uri)
       uri
     else
       target = res.target.to_s.strip
+      target_host = URI("https://" + target).host
 
-      if URI("http://" + target).host.end_with?(".#{host}")
-        return URI.parse "#{uri.scheme}://#{target}#{uri.path}"
+      if target_host.end_with?(".#{host}")
+        uri = uri.dup
+        uri.host = target_host
+        return uri
```

Comment:
this `return uri` is redundant it's already returned 3 lines below

Reply:
Now that i think about it more, I'm thinking the assignment of the #dup call should be something other than uri, like modified_uri, to make it more clear that we're returning something other than the original uri.  That probably makes this more relevant and less likely to stop the uri variable name in this context.